develop/parachain-devs/runtime-development/FRAME/create-custom-pallet.md

```diff
@@ -0,0 +1,165 @@
+    --8<-- 'code/develop/parachain-devs/runtime-development/FRAME/create-custom-pallet/Cargo.toml'
+    ```
+
+    !!!note
+        Proper version management is crucial for ensuring compatibility and reducing potential conflicts in your project. Carefully select the versions of the packages according to your project's specific requirements:
+
+        - When developing for a specific Polkadot SDK runtime, ensure that your pallet's dependency versions match those of the target runtime
+        - If you're creating this pallet within a Polkadot SDK workspace:
+
+            - Define the actual versions in the root `Cargo.toml` file
+            - Use workspace inheritance in your pallet's `Cargo.toml` to maintain consistency across your project
+        - Regularly check for updates to FRAME and Polkadot SDK dependencies to benefit from the latest features, performance improvements, and security patches
+
+        For more detailed information on workspace inheritance and how to properly integrate your pallet with the runtime, refer to the [Add Existing Pallet to the Runtime](TODO:update-path){target=\_blank} page.
+
+3.  Set up the basic pallet structure
+
+    After setting up the project and configuring dependencies, the next step is to prepare the basic structure of your pallet in the `lib.rs` file. Open the `src/lib.rs` file in your project, remove all existing content and copy and paste the following scaffold code:
+
+    ```rust
+    --8<-- 'code/develop/parachain-devs/runtime-development/FRAME/create-custom-pallet/barebone-pallet-template.rs'
+    ```
+
+    With this scaffold in place, you're ready to start implementing the specific logic and features of your custom pallet. The subsequent sections of this guide will walk you through populating each of these components with the necessary code for your pallet's functionality.
+
+## Pallet Configuration
+
+Every pallet includes a Rust trait called [`Config`](https://paritytech.github.io/polkadot-sdk/master/frame_system/pallet/trait.Config.html){target=\_blank}, which is used to expose configurable options and link your pallet to other parts of the runtime. All types and constants that the pallet depends on must be declared within this trait. These types are defined generically and made concrete when the pallet is instantiated in the `runtime/src/lib.rs` file of your blockchain.
+
+In this step, you'll only configure the common types used by all pallets:
+
+- `RuntimeEvent` - since this pallet emits events, it requires the runtime's event type to handle them. This ensures that events generated by the pallet can be correctly processed and interpreted by the runtime
+- `WeightInfo` - this type defines the weights associated with the pallet's callable functions (also known as dispatchables). Weights help measure the computational cost of executing these functions. However the WeightInfo type will be left unconfigured, since setting up custom weights is outside the scope of this guide
```

Comment:
```suggestion
- **`RuntimeEvent`** - since this pallet emits events, the runtime event type is required to handle them. This ensures that events generated by the pallet can be correctly processed and interpreted by the runtime
- **`WeightInfo`** - this type defines the weights associated with the pallet's callable functions (also known as dispatchables). Weights help measure the computational cost of executing these functions. However, the `WeightInfo` type will be left unconfigured since setting up custom weights is outside the scope of this guide
```